src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,156 @@
+                book.setTitle(title);
+                book.setPrice(price);
+                return Optional.of(book);
+            }
+
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't create connection", e);
+        }
+        return Optional.empty();
+    }
+
+    @Dao
+    @Override
+    public List<Book> findAll() {
+        Long id = 0L;
+        List<Book> bookList = new ArrayList<>();
+
+        String sql = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet;
+
+            do {
+                resultSet = preparedStatement.executeQuery();
+                Long rowId = resultSet.getObject("id", Long.class);
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getObject("price", BigDecimal.class);
+
+                Book book = new Book();
+                book.setId(rowId);
+                book.setTitle(title);
+                book.setPrice(price);
+
+                bookList.add(book);
+                id++;
+            } while (resultSet.next());
+
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't create connection", e);
+        }
+        return bookList;
+    }
+
+    @Dao
+    @Override
+    public Book update(Book book) {
+        String sql = "SELECT * FROM books WHERE title = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement
+                        = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS);) {
+            preparedStatement.setString(1, book.getTitle());
+            int affectedRows = preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getResultSet();
+            if (resultSet.next()) {
+                book.setPrice(book.getPrice());
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't create new book: " + book, e);
+        }
+    }
+
+    @Dao
+    @Override
+    public boolean deleteById(Long id) {
+        String sql = "DELETE FROM books WHERE id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement
+                        = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS);) {
+            preparedStatement.setLong(1, id);
+            int affectedRows = preparedStatement.executeUpdate();
+            try {
+                boolean b = affectedRows < 1;
+            } catch (DataProcessingException ex) {
+                throw new DataProcessingException("Expected to delete at least one row, "
+                        + "but deleted 0 rows.", ex);
+            }

```

Comment:
Similar to the `create` method, the try-catch block here is incorrectly used. Directly check if `affectedRows < 1` and handle the situation appropriately without using a try-catch block for this condition.

Reply:
ok, done